src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -260,7 +260,9 @@ internal override bool TryGetSpeculativeSemanticModelForMethodBodyCore(SyntaxTre
 
         protected override BoundNode RewriteNullableBoundNodesWithSnapshots(BoundNode boundRoot, Binder binder, DiagnosticBag diagnostics, bool createSnapshots, out NullableWalker.SnapshotManager snapshotManager)
         {
-            return NullableWalker.AnalyzeAndRewrite(Compilation, MemberSymbol, boundRoot, binder, diagnostics, createSnapshots, out snapshotManager);
+            var symbol = boundRoot is BoundPropertyEqualsValue { Property: SourcePropertySymbol { BackingField: var backingField } } ? backingField : MemberSymbol;
```

Comment:
This is necessary because the initializer's containing symbol is the backing field, not the property.